First android build failed

### Steps to reproduce

1. git checkout 827b11c764d63123259560e8546aab71b61c1470 - current master
2. dart run /patrol/packages/patrol_cli/bin/main.dart test --verbose --target integration_test/example_test.dart

### Actual results

Android test builds always failed with this error. Some solutions from the Internet didn't help, but the problem disappeared when I built the example app normally using "flutter run". This issue is similar to the other where we need to build our app at least once. 

### Logs

<details>
<summary>Logs</summary>

```
	FAILURE: Build failed with an exception.
	
	* What went wrong:
	A problem occurred configuring root project 'android'.
	> Could not open proj generic class cache for build file 'patrol/packages/patrol/example/android/build.gradle' (/.gradle/caches/7.6.2/scripts/7jd035x0svt1nqxqk95ifino8).
	   > BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 64
	
	* Try:
	> Run with --stacktrace option to get the stack trace.
	> Run with --info or --debug option to get more log output.
	> Run with --scan to get full insights.
	
	* Get more help at https://help.gradle.org
	
	BUILD FAILED in 619ms
✗ Failed to build apk with entrypoint test_bundle.dart (Gradle build failed with code 1) (1.1s)
```

</details>


### Patrol version

patrol: master == 827b11c764d63123259560e8546aab71b61c1470


### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
You're using Patrol CLI 2.0, which has breaking changes.
Read the migration guide at https://patrol.leancode.co/v2.
Disable this warning by setting the PATROL_MIGRATED environment variable.

Patrol CLI version: 2.1.1
Program adb found in /Users/User/Library/Android/sdk/platform-tools/adb
Env var $ANDROID_HOME set to /Users/User/Library/Android/sdk
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller not found (install with `brew install ideviceinstaller`)
```

</details>


### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.4.1 22F770820d darwin-x64, locale pl-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.81.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!

```

</details>

--- a/packages/patrol_cli/CHANGELOG.md
+++ b/packages/patrol_cli/CHANGELOG.md
@@ -1,5 +1,6 @@
 ## 3.1.0
 - Add `tags` and `exclude-tags`. (#2286)
+- Run `flutter build apk --config-only` during android build.(#2293)
 
 This version requires version 3.10.0 of `patrol` package.
 

--- a/packages/patrol_cli/lib/src/android/android_test_backend.dart
+++ b/packages/patrol_cli/lib/src/android/android_test_backend.dart
@@ -41,6 +41,7 @@ class AndroidTestBackend {
   late final String? javaPath;
 
   Future<void> build(AndroidAppOptions options) async {
+    await buildApkConfigOnly(options.flutter.command.executable);
     await loadJavaPathFromFlutterDoctor(options.flutter.command.executable);
 
     await _disposeScope.run((scope) async {
@@ -147,6 +148,21 @@ class AndroidTestBackend {
     });
 
     javaPath = await javaCompleterPath.future;
+  }
+
+  /// Execute `flutter build apk --config-only` to generate the gradlew file.
+  ///
+  /// This fix issue: https://github.com/leancodepl/patrol/issues/1668
+  Future<void> buildApkConfigOnly(String commandExecutable) async {
+    await _processManager.start(
+      [
+        commandExecutable,
+        'build',
+        'apk',
+        '--config-only',
+      ],
+      runInShell: true,
+    );
   }
 
   /// Executes the tests of the given [options] on the given [device].

